### PR TITLE
Insert sorted vertices in batch for tangential complex

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Fix partially #718 for the tangential complex

### CODE REVIEW

**Comment by VincentRouvreau** [comment]
```python
import gudhi as gd
from gudhi.datasets.generators import points
data = points.sphere(n_samples = 500000, ambient_dim = 2, radius = 1000., sample = "random")
tc = gd.TangentialComplex(intrisic_dim = 1,points=data)
tc.compute_tangential_complex()
result_str = 'Tangential contains ' + repr(tc.num_simplices()) + \
    ' simplices - ' + repr(tc.num_vertices()) + ' vertices.'
print(result_str)
# Tangential contains 229123 simplices - 114562 vertices.
from timeit import default_timer as timer
from datetime import timedelta
start = timer(); st = tc.create_simplex_tree(); print(timedelta(seconds=timer()-start))
# 0:00:24.924412 (master @ 5d0c294572a9e34a7a70e78806ac3cf805357712) vs 0:00:00.391726 (PR #762)
```

File: src/Tangential_complex/include/gudhi/Tangential_complex.h [hunk 1]
```diff
@@ -624,6 +625,12 @@ class Tangential_complex {
 
     int max_dim = -1;
 
+    // Vertices to be inserted first by the create_complex method to avoid quadratic complexity.
+    // It isn't just [0, n) if some points have multiplicity (only one copy appears in the complex).
+    std::vector<typename Simplex_tree_::Vertex_handle> vertices(m_points.size());
+    std::iota(vertices.begin(), vertices.end(), 0);
```
Comment (mglisse) [inline src/Tangential_complex/include/gudhi/Tangential_complex.h:631]: The comment does not match the code at all.

**Review by mglisse — COMMENTED** [review]
I had forgotten that the tangential complex required a simplex tree instead of some concept like the others. At least here it means we don't need to modify the concept :-)

File: src/Tangential_complex/include/gudhi/Tangential_complex.h (hunk 1 above)
Comment (VincentRouvreau) [reply src/Tangential_complex/include/gudhi/Tangential_complex.h]: Bad copy/paste of a comment. Fixed on [6b8da55](https://github.com/GUDHI/gudhi-devel/pull/762/commits/6b8da55930c5a76aefc6725d3cffb17bab5438d0)

**Review by mglisse — APPROVED** [review]

**Pull request merged**